build/patches/experimental-user-scripts-support.patch

```diff
@@ -0,0 +1,10299 @@
+     "//components/dom_distiller/content/renderer",
++    "//components/user_scripts/renderer",
+     "//components/network_hints/renderer",
+     "//components/no_state_prefetch/common",
+     "//components/no_state_prefetch/renderer",
+diff --git a/chrome/renderer/chrome_content_renderer_client.cc b/chrome/renderer/chrome_content_renderer_client.cc
+--- a/chrome/renderer/chrome_content_renderer_client.cc
++++ b/chrome/renderer/chrome_content_renderer_client.cc
+@@ -166,6 +166,8 @@
+ 
+ #if defined(OS_ANDROID)
+ #include "chrome/renderer/sandbox_status_extension_android.h"
++#include "components/user_scripts/common/user_scripts_features.h"
++#include "components/user_scripts/renderer/user_scripts_renderer_client.h"
+ #else
+ #include "chrome/renderer/media/chrome_speech_recognition_client.h"
+ #include "chrome/renderer/searchbox/searchbox.h"
+@@ -361,6 +363,12 @@ void ChromeContentRendererClient::RenderThreadStarted() {
+   ChromeExtensionsRendererClient::GetInstance()->RenderThreadStarted();
+ #endif
+ 
++  user_scripts::UserScriptsRendererClient* userscript_client =
++    user_scripts::UserScriptsRendererClient::GetInstance();
++  if(userscript_client) {
++    userscript_client->RenderThreadStarted();
++  }
++
+ #if BUILDFLAG(ENABLE_SPELLCHECK)
+   if (!spellcheck_)
+     InitSpellCheck();
+@@ -497,6 +505,13 @@ void ChromeContentRendererClient::RenderFrameCreated(
+       render_frame, registry);
+ #endif
+ 
++  user_scripts::UserScriptsRendererClient* userscript_client =
++    user_scripts::UserScriptsRendererClient::GetInstance();
++  if(userscript_client) {
++    userscript_client->RenderFrameCreated(
++      render_frame, registry);
++  }
++
+ #if BUILDFLAG(ENABLE_PLUGINS)
+   new PepperHelper(render_frame);
+ #endif
+@@ -1470,6 +1485,12 @@ void ChromeContentRendererClient::RunScriptsAtDocumentStart(
+       render_frame);
+   // |render_frame| might be dead by now.
+ #endif
++  user_scripts::UserScriptsRendererClient* userscript_client =
++    user_scripts::UserScriptsRendererClient::GetInstance();
++  if(userscript_client) {
++    userscript_client->RunScriptsAtDocumentStart(
++      render_frame);
++  }
+ }
+ 
+ void ChromeContentRendererClient::RunScriptsAtDocumentEnd(
+@@ -1479,6 +1500,12 @@ void ChromeContentRendererClient::RunScriptsAtDocumentEnd(
+       render_frame);
+   // |render_frame| might be dead by now.
+ #endif
++  user_scripts::UserScriptsRendererClient* userscript_client =
++    user_scripts::UserScriptsRendererClient::GetInstance();
++  if(userscript_client) {
++    userscript_client->RunScriptsAtDocumentEnd(
++      render_frame);
++  }
+ }
+ 
+ void ChromeContentRendererClient::RunScriptsAtDocumentIdle(
+@@ -1488,6 +1515,12 @@ void ChromeContentRendererClient::RunScriptsAtDocumentIdle(
+       render_frame);
+   // |render_frame| might be dead by now.
+ #endif
++  user_scripts::UserScriptsRendererClient* userscript_client =
++    user_scripts::UserScriptsRendererClient::GetInstance();
++  if(userscript_client) {
```

Comment:
```suggestion
+  if (userscript_client) {
```

Reply:
Shouldn't we also check that `render_frame` is not dead by now? See upstream comment above.

Reply:
BUILDFLAG(ENABLE_EXTENSIONS) need to be not enabled.
if it were active our patch would not work and therefore it is right for it to crash.
I prefer an application to crash rather than "not" run for no reason.

Reply:
Are you sure that it will not be `null` in any other case?

Reply:
add `static_assert` in
https://github.com/bromite/bromite/blob/8c164138391082350f20493a0ca5c4926887372c/build/patches/experimental-user-scripts-support.patch#L598